doc: reword announcement about upcoming 1.0.0

lukpueh opened the pull request:
NOTE: Should be merged directly before release (coordinate with #1851)

Fixes #1795 

**Description of the changes being introduced by the pull request**:
Change docs in preparation of close v1.0.0 release.

- Remove important notice about upcoming 1.0.0 release from README
- Reword 1.0.0-ANNOUNCEMENT.md to not sound outdated after release

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
doc: reword announcement about upcoming 1.0.0

Change docs in preparation of close v1.0.0 release.

- Remove important notice about upcoming 1.0.0 release from README
- Reword 1.0.0-ANNOUNCEMENT.md to not sound outdated after release

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

joshuagl commented on docs/1.0.0-ANNOUNCEMENT.md:
```diff
@@ -1,8 +1,7 @@
-python-tuf reference implementation, and we are very excited to announce a
-stable 1.0.0 release scheduled for January 2022. The release *will* include:
+Python-TUF v1.0.0 is a rewritten stable reference implementation of the TUF
+specification, which *includes*:
```
```suggestion
specification, which *currently* includes:
```
opposite to the "*does not yet* include a new *repository tool*" statement

joshuagl commented on docs/1.0.0-ANNOUNCEMENT.md:
```diff
@@ -11,7 +10,7 @@ stable 1.0.0 release scheduled for January 2022. The release *will* include:
   for the legacy client updater
 
 As discussed in [ADR 2](https://github.com/theupdateframework/python-tuf/blob/develop/docs/adr/0002-pre-1-0-deprecation-strategy.md), this
-release *will not* include any legacy code, as its maintenance has become
+release *does not* include any legacy code, as its maintenance has become
 infeasible for the python-tuf team. The pre-1.0.0 deprecation strategy from ADR
 2 applies as follows:
```
Now that this code has been removed, should this statement be moved down? The next paragraph seems more important.

joshuagl commented on docs/1.0.0-ANNOUNCEMENT.md:
```diff
@@ -22,13 +21,11 @@ processes). Note that there may be delays due to the lack of developer resources
-), this release *will not yet* include a new *repository tool*. However, the new
+), this release *does not yet* include a new *repository tool*. However, the new
 *metadata API* makes it easy to replicate the desired functionality tailored to
 the specific needs of any given repository (see *Migration* for details).
```
```suggestion
), this release *does not yet* include *repository tool*-like functionality. However, the new
*metadata API* makes it easy to replicate the desired functionality tailored to
the specific needs of any given repository (see *Migration* for details).
```
Nit: we don't want to create a new repository-tool, we want to create something better. This edit tries to emphasise that we want to provide repository-tool-like functionality

joshuagl reviewed (commented):
Thanks. This looks good. I made a few minor suggestions, WDYT?

lukpueh committed:
doc: tweak announcement about upcoming 1.0.0

Further tweak doc in preparation of close v1.0.0 release:

- Minor rewords/re-formats
- Swap paragraphs according to importance

Co-authored-by: Joshua Lock <jlock@vmware.com>
Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
Thanks for the review, @joshuagl! All your suggestions make sense and are now in https://github.com/theupdateframework/python-tuf/pull/1869/commits/399a4823b49b2ec7232c75bc1eb225d2d831b6a8

joshuagl reviewed (approved):
Thanks for the quick edits, LGTM

lukpueh commented:
Superseded by #1873
